pkg/certificate/providers/certmanager/certificate_manager.go

```diff
@@ -149,19 +41,12 @@ func (cm *CertManager) certificateFromCertificateRequest(cr *cmapi.CertificateRe
 	}, nil
 }
 
-// issue will request a new signed certificate from the configured cert-manager
-// issuer.
-func (cm *CertManager) issue(cn certificate.CommonName, validityPeriod time.Duration) (*certificate.Certificate, error) {
+// IssueCertificate will request a new signed certificate from the configured cert-manager issuer.
+func (cm *CertManager) IssueCertificate(cn certificate.CommonName, validityPeriod time.Duration) (*certificate.Certificate, error) {
 	duration := &metav1.Duration{
 		Duration: validityPeriod,
 	}
 
-	// Key bit size should remain static during the lifetime of the CertManager. In the event that this
-	// is a zero value, we make the call to config to get the setting and then cache it for future
-	// certificate operations.
-	if cm.keySize == 0 {
```

Comment:
Where is this validation being performed now?

Reply:
It was essentially an erroneous validation in the first place. Since it was always initially set from `cm.cfg.GetCertKeyBitSize()`, and if that's zero it would reset to cm.cfg.GetCertKeyBitSize(), which still may be zero. Eventually this would result in an error regardless of this check.

Ideally we add this to either a) validating webhook, b) give a hard error on instantiation, or c) both a,b.

for now i re-added back to the instantiation of the client

Reply:
This is not an erroneous validation. The reason it was done this way was to allow instantiating the certificate provider instances outside the osm-controller (e.g. from the CLI) with a specified key size. There was a use case to be able to issue a certificate on-demand using OSM CLI for ingress at some point which would make use of the CLI to issue a cert.